src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -42,9 +43,21 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
         model.deletePerson(personToDelete);
+        updateLastViewedPersonIfNecessary(personToDelete, model);
+
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
     }
 
+    private void updateLastViewedPersonIfNecessary(Person personToDelete, Model model) {
+        Optional<Person> lastViewedPerson = model.getLastViewedPerson();
+        if (lastViewedPerson.isEmpty()) {
+            return;
+        }
+        if (lastViewedPerson.get().equals(personToDelete)) {
+            model.resetLastViewedPerson();
+        }
+    }
+
```

Comment:
Code looks great, but I thought you could perhaps make use of the Optional class and avoid explicit checks. Something like the following:
```
private void updateLastViewedPersonIfNecessary(Person personToDelete, Model model) {
    model.getLastViewedPerson()
         .filter(lastViewedPerson -> lastViewedPerson.equals(personToDelete))
         .ifPresent(lastViewedPerson -> model.resetLastViewedPerson());
}
```

Reply:
Can be shortened to 

```java
private void updateLastViewedPersonIfNecessary(Person personToDelete, Model model) {
    model.getLastViewedPerson()
         .filter(lastViewedPerson -> lastViewedPerson.equals(personToDelete))
         .ifPresent(model::resetLastViewedPerson());
}
```

Reply:
Agreed, made the change.